docs/developers/piece-reference/properties.mdx

```diff
@@ -334,3 +334,179 @@ Property.DynamicProperties({
     }
 })
 ```
+
+## Validators and Processors
+The Validators and Processors feature offers you the means to optimize data quality by efficiently processing user input and validating it accordingly. This section provides comprehensive guidance on how to effectively utilize these functionalities.
+
+### PieceProperty Type Changes
+The following four keys have been added to the `PieceProperty` type:
+
+```typescript
+defaultProcessors?: ProcessorFn[];
+defaultValidators?: TypedValidatorFn<ValidationInputType>[];
+processors?: ProcessorFn[];
+validators?: TypedValidatorFn<ValidationInputType>[];
+```
+
+- `defaultProcessors`: Represents an array of processor functions that will always be applied to the property and cannot be overwritten.
+- `defaultValidators`: Represents an array of validator functions that will always be applied to the property and cannot be overwritten.
+- `processors`: Represents an optional array of processor functions that can be passed when defining the property.
+- `validators`: Represents an optional array of validator functions that can be passed when defining the property.
+
+### How to Use
+To apply `defaultProcessors` and `defaultValidators` to a property, you can use them in the constructor as shown in the example below:
+
+```typescript
+DateTime<R extends boolean>(request: Properties<DateTimeProperty<R>>): R extends true ? DateTimeProperty<true> : DateTimeProperty<false> {
+    return { 
+        ...request, 
+        defaultProcessors: [Processors.datetime],
+        defaultValidators: [Validators.datetimeIso],
+        valueSchema: undefined, 
+        type: PropertyType.DATE_TIME, 
+    } as unknown as R extends true ? DateTimeProperty<true> : DateTimeProperty<false>;
+}
+```
+
+On the other hand, if you want to specify custom `processors` and `validators` for a property, you can pass them in the property definition, as demonstrated in the example below:
+
+```typescript
+text: Property.LongText({
+    displayName: 'Text',
+    required: true,
+    validators: [Validators.url, Validators.oneOf(["activepieces.com", "www.activepieces.com"])],
+})
+```
+
+### Processors
```

Comment:
I would move everything related to processor under this section even custom processor, so it's self contained

Reply:
I would suggest the following as well, so it shows the outline in better way

![image](https://github.com/activepieces/activepieces/assets/1812998/17420f65-d72c-4556-8f9a-97e61a426fc7)

1- Change the Header to ## 
2- Remove Predefined Processors Headers
3- Each Predefined Processor should be ### So it shows in the outline.
4- and as well move the custom processor as one of the  ###  here,
### File 
### DateTime
### Number

Same to the validators.